src/expr/type_node.cpp

```diff
@@ -201,6 +202,37 @@ CardinalityClass TypeNode::getCardinalityClass()
   return ret;
 }
 
+bool TypeNode::isCardinalityLessThan(size_t n)
+{
+  if ((isDatatype() && !isInstantiated())
```

Comment:
You can delete this block.

Reply:
You mean the guard for instantiated datatypes? I cannot, this fails on our regressions otherwise.